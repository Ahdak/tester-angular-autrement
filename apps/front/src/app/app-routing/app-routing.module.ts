import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { ListComponent } from '../rentals/list/list.component';
import { NotfoundPageComponent } from '../shell/pages/404/notfound-page.component';
import { DetailsComponent } from '../rentals/details/details.component';
import { LoginPageComponent } from '../auth/login-page/login-page.component';
import { AuthGuard } from '../auth/auth.guard';
import { DetailsGuard } from '../rentals/details/details.guard';

export const routes: Routes = [
  { path: '', redirectTo: '/rentals', pathMatch: 'full' },
  { path: 'login', component: LoginPageComponent },
  { path: 'rentals', component: ListComponent },
  {
    path: 'rentals/:id',
    component: DetailsComponent,
    canActivate: [AuthGuard, DetailsGuard],
  },
  { path: '**', component: NotfoundPageComponent },
];

@NgModule({
  imports: [
    RouterModule.forRoot(routes, {
      useHash: true,
      scrollPositionRestoration: 'top',
    }),
  ],
  exports: [RouterModule],
})
export class AppRoutingModule {}
