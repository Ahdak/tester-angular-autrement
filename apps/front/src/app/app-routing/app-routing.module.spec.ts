import { AppRoutingModule, routes } from './app-routing.module';
import { Route } from '@angular/router';

expect.addSnapshotSerializer({
  test: (confRoutes: Route[]) => {
    const isRoot = (object: any): object is Route => {
      return 'path' in object;
    };
    return confRoutes.reduce((acc, curr) => acc && isRoot(curr), true);
  },
  print: (confRoutes: Route[]) => {
    let str = 'Routes: [ \n';
    str += confRoutes.map((route) => {
      return `   Route: {
            path: ${route.path},
            component: ${route.component ? route.component.prototype.constructor.name : undefined},
            redirectTo: ${route.redirectTo},
            pathMatch: ${route.pathMatch},
            canActivate: ${route.canActivate ? route.canActivate.map((guard) => {
        return guard.prototype.constructor.name;
      }).join(',') : undefined}
          }\n`;
    });
    str += ']';
    return str;
  }
});

describe('AppRoutingModule', () => {
  let routingModule: AppRoutingModule;

  beforeEach(() => {
    routingModule = new AppRoutingModule();
  });

  it('should create an instance', () => {
    expect(routingModule).toBeTruthy();
  });

  it('should contains correct routes/guards', () => {
    expect(routes).toMatchSnapshot();
  });

});
