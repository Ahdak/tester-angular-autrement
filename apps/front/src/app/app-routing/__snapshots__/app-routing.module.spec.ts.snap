// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoutingModule should contains correct routes/guards 1`] = `
Routes: [ 
   Route: {
            path: ,
            component: undefined,
            redirectTo: /rentals,
            pathMatch: full,
            canActivate: undefined
          }
,   Route: {
            path: login,
            component: LoginPageComponent,
            redirectTo: undefined,
            pathMatch: undefined,
            canActivate: undefined
          }
,   Route: {
            path: rentals,
            component: ListComponent,
            redirectTo: undefined,
            pathMatch: undefined,
            canActivate: undefined
          }
,   Route: {
            path: rentals/:id,
            component: DetailsComponent,
            redirectTo: undefined,
            pathMatch: undefined,
            canActivate: AuthGuard,DetailsGuard
          }
,   Route: {
            path: **,
            component: NotfoundPageComponent,
            redirectTo: undefined,
            pathMatch: undefined,
            canActivate: undefined
          }
]
`;
